TROUBLESHOOTING.md

```diff
@@ -36,7 +36,42 @@ to index for features such as go to definition.
 
 ## Common issues
 
-There are two main sources of issues users typically face during activation: shell or Bundler related problems.
+There are three main sources of issues users typically face during activation: outdated version, shell problems, or Bundler related problems.
+
+### Outdated Version
+
+In some circumstances, the version of Ruby LSP activated may be very outdated. There are two common reasons for this:
+
+**Prism constraints**
+
+Since v0.12.0, Ruby LSP has had a dependency on the [Prism](https://rubygems.org/gems/prism) parser.
+Prior to that, the Prism parser was named [YARP](https://rubygems.org/gems/yarp). The fact that the gem was renamed leads to some awkward dependency resolutions issues.
+Since Prism is a pre-1.0 release, there may be breaking changes introduced in minor versions.
+For that reason, we constrain the version of Prism up to that which is known to be compatible.
+
+With the custom bundle approach describe earlier, Bundler resolves a version of Ruby LSP which is compatible the dependencies already in your `Gemfile.lock.`
+
+When a new version of Prism is released, it will take a little time for us as Ruby LSP maintainers to verify the compatibility, and make any necessary updates.
+
+During that time, it's possible for the Prism version in your `Gemfile.lock` to be increased due to being a dependency of another gem in your bundle.
+
+If Prism is locked to a newer version not yet supported by the Ruby LSP, then the only outcome that satisfies Bundler checks is to install the very old versions that depended on the old name YARP.
+
+**How to solve it**
+
+You can add a dependency constraint to your Gemfile limiting the Prism version to that maximum supported by Ruby LSP.
+For example:
+
+```
+gem "prism", "< 0.31" # example for Ruby LSP v0.17.7
```

Comment:
I feel this will bite us once Prism v1.0 is out. Since the best solution is for Ruby LSP to release a version that support the latest Prism, let's not suggest this workaround?